src/servers/main.cc

```diff
@@ -60,26 +60,44 @@ std::condition_variable exit_cv_;
 // exit if inference-server doesn't completely initialize (e.g. will
 // exit if even one model fails to load).
 bool exit_on_failed_init_ = true;
+constexpr size_t unique_service_endpoint_count_ = 4;
 
 // The HTTP and GRPC services
-std::unique_ptr<nvidia::inferenceserver::HTTPServer> http_service_;
 std::unique_ptr<nvidia::inferenceserver::GRPCServer> grpc_service_;
```

Comment:
Why are there still the individual http_server_ and grpc_server_?

Reply:
This is for when the http-port is set to some value. If it is set to -1 then the multi-port option is triggered

Reply:
Don't think about 2 different modes. There is only one mode that has 4 inputs (the ports). There are 5 command-line flags (plus the allow flag) that are used to initialize those 4 inputs.

Reply:
Alright. Will combine the functionality to support the above.